osu.Game/Screens/Play/Player.cs

```diff
@@ -979,6 +1002,19 @@ protected virtual async Task ImportScore(Score score)
             score.ScoreInfo.OnlineScoreID = onlineScoreId;
         }
 
+        /// <summary>
+        /// Prepare the <see cref="Scoring.Score"/> for display at results.
+        /// </summary>
+        /// <remarks>
+        /// This is run synchronously before <see cref="PrepareScoreForResultsAsync"/> is run.
+        /// </remarks>
+        /// <param name="score">The <see cref="Scoring.Score"/> to prepare.</param>
+        protected virtual void PrepareScoreForResults(Score score)
+        {
+            // perform one final population to ensure everything is up-to-date.
+            ScoreProcessor.PopulateScore(score.ScoreInfo);
+        }
```

Comment:
I propose removing this method and only keeping the `async` variety around.

Reply:
Definitely agree with keeping async only unless there's a very good reason not to, as having both already confused me when trying to read this. If someone wants a sync operation there then there is always `Task.FromResult()`, using which should make `await`s run the logic inline synchronously.